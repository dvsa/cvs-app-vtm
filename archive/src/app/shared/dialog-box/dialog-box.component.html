<div mat-dialog-content>
  <h2 class="govuk-heading-l">{{ data.context }}</h2>
  <textarea
    class="govuk-textarea govuk-!-width-three-quarters"
    id="test-reasonForCreation"
    rows="8"
    [(ngModel)]="data.response"
    aria-describedby="reasonForCreation"
  >
  </textarea>
</div>

<div class="govuk-grid-row">
  <div class="govuk-grid-column-one-quarter govuk-!-margin-right-0 govuk-!-padding-right-0">
    <button
      class="govuk-button"
      id="test-{{ data.actionName }}"
      data-module="govuk-button"
      (click)="save(data.response)"
      cdkFocusInitial
    >
      {{ data.actionName }}
    </button>
  </div>
  <div
    class="govuk-grid-column-one-quarter govuk-!-margin-right-0 govuk-!-padding-right-0 govuk-!-padding-top-2"
  >
    <a class="govuk-link govuk-!-margin-left-2" (click)="close()" href="javascript:void(0)">
      <span>Cancel</span>
    </a>
  </div>
</div>
