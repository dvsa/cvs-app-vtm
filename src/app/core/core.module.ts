import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { GlobalErrorComponent } from './components/global-error/global-error.component';
import { SpinnerComponent } from './components/spinner/spinner.component';
import { FooterComponent } from './components/footer/footer.component';
import { HeaderComponent } from './components/header/header.component';
import { RouterModule } from '@angular/router';
import { BreadcrumbsComponent } from './components/breadcrumbs/breadcrumbs.component';
import { PageNotFoundComponent } from './components/page-not-found/page-not-found.component';
import { ServerErrorComponent } from './components/server-error/server-error.component';

@NgModule({
  declarations: [FooterComponent, HeaderComponent, GlobalErrorComponent, SpinnerComponent, BreadcrumbsComponent, PageNotFoundComponent, ServerErrorComponent],
  imports: [CommonModule, RouterModule],
  exports: [FooterComponent, HeaderComponent, GlobalErrorComponent, SpinnerComponent, BreadcrumbsComponent]
})
export class CoreModule {}
