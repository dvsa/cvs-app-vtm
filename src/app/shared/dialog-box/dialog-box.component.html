<div mat-dialog-content>
  <h2 class="govuk-heading-l">{{ data.context }}</h2>
  <textarea
    class="govuk-textarea govuk-!-width-three-quarters"
    id="reasonForCreation"
    rows="8"
    [(ngModel)]="data.response"
    aria-describedby="reasonForCreation"
  >
  </textarea>
</div>

<button
  class="govuk-button"
  data-module="govuk-button"
  (click)="save(data.response)"
  cdkFocusInitial
>
  {{data.actionName}}
</button>

<a class="govuk-link govuk-!-margin-left-2 link--cancel" (click)="close()">
  <span>Cancel</span>
</a>
