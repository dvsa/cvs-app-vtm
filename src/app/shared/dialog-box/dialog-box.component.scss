textarea {
  width: 80% !important;
}

.cancel {
  color: #005EA5;
  font-family: "GDS Transport", Arial, sans-serif;
  font-size: 19px;
  font-weight: 300;
  line-height: 44px;
  text-align: left;
  padding-left: 2rem;
  cursor: pointer;
  span {
    border-bottom: solid;
    border-bottom-width: 2px;
  }
}
