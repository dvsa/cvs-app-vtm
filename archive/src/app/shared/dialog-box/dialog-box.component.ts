import { Component, OnInit, ChangeDetectionStrategy, Inject } from '@angular/core';
import { MAT_DIALOG_DATA, MatDialogRef } from '@angular/material/dialog';
import { DialogData } from '@app/models/dialog-data';

@Component({
  selector: 'vtm-dialog-box',
  templateUrl: './dialog-box.component.html',
  changeDetection: ChangeDetectionStrategy.OnPush
})
export class DialogBoxComponent implements OnInit {
  constructor(
    public dialogRef: MatDialogRef<DialogBoxComponent>,
    @Inject(MAT_DIALOG_DATA) public data: DialogData
  ) {}

  ngOnInit() {}

  close(): void {
    this.dialogRef.close(false);
  }

  save(modalData: string): void {
    this.dialogRef.close({ isSave: true, data: modalData });
  }
}
