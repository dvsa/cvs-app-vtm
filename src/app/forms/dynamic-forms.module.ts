import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { FormsModule, ReactiveFormsModule } from '@angular/forms';
import { RouterModule } from '@angular/router';
import { NoSpaceDirective } from '@directives/app-no-space/app-no-space.directive';
import { NumberOnlyDirective } from '@directives/app-number-only/app-number-only.directive';
import { ToUppercaseDirective } from '@directives/app-to-uppercase/app-to-uppercase.directive';
import { TrimWhitespaceDirective } from '@directives/app-trim-whitespace/app-trim-whitespace.directive';
import { ApprovalTypeFocusNextDirective } from '@directives/approval-type-focus-next/approval-type-focus-next.directive';
import { DateFocusNextDirective } from '@directives/date-focus-next/date-focus-next.directive';
import { PrefixDirective } from '@directives/prefix/prefix.directive';
import { SuffixDirective } from '@directives/suffix/suffix.directive';
import { ApprovalTypeInputComponent } from '@forms/components/approval-type/approval-type.component';
import { AdrCertificateHistoryComponent } from '@forms/custom-sections/adr-certificate-history/adr-certificate-history.component';
import { AdrExaminerNotesHistoryEditComponent } from '@forms/custom-sections/adr-examiner-notes-history-edit/adr-examiner-notes-history.component-edit';
import { AdrPermittedDangerousGoodsComponent } from '@forms/custom-sections/adr-permitted-dangerous-goods/adr-permitted-dangerous-goods.component';
import { ApprovalTypeComponent } from '@forms/custom-sections/approval-type/approval-type.component';
import { SharedModule } from '@shared/shared.module';
import { TruncatePipe } from '../pipes/truncate/truncate.pipe';
import { AutocompleteComponent } from './components/autocomplete/autocomplete.component';
import { BaseControlComponent } from './components/base-control/base-control.component';
import { CharacterCountComponent } from './components/character-count/character-count.component';
import { CheckboxGroupComponent } from './components/checkbox-group/checkbox-group.component';
import { CheckboxComponent } from './components/checkbox/checkbox.component';
import { ContingencyAdrGenerateCertComponent } from './components/contingency-adr-generate-cert/contingency-adr-generate-cert.component';
import { ControlErrorsComponent } from './components/control-errors/control-errors.component';
import { DateControlsComponent } from './components/date-controls/date-controls.component';
import { DateComponent } from './components/date/date.component';
import { DefectSelectComponent } from './components/defect-select/defect-select.component';
import { DynamicFormFieldComponent } from './components/dynamic-form-field/dynamic-form-field.component';
import { DynamicFormGroupComponent } from './components/dynamic-form-group/dynamic-form-group.component';
import { FieldErrorMessageComponent } from './components/field-error-message/field-error-message.component';
import { FieldWarningMessageComponent } from './components/field-warning-message/field-warning-message.component';
import { NumberInputComponent } from './components/number-input/number-input.component';
import { RadioGroupComponent } from './components/radio-group/radio-group.component';
import { ReadOnlyComponent } from './components/read-only/read-only.component';
import { RequiredStandardSelectComponent } from './components/required-standard-select/required-standard-select.component';
import { SelectComponent } from './components/select/select.component';
import { SuggestiveInputComponent } from './components/suggestive-input/suggestive-input.component';
import { SwitchableInputComponent } from './components/switchable-input/switchable-input.component';
import { TextAreaComponent } from './components/text-area/text-area.component';
import { TextInputComponent } from './components/text-input/text-input.component';
import { ViewCombinationComponent } from './components/view-combination/view-combination.component';
import { ViewListItemComponent } from './components/view-list-item/view-list-item.component';
import { AbandonDialogComponent } from './custom-sections/abandon-dialog/abandon-dialog.component';
import { AdrExaminerNotesHistoryViewComponent } from './custom-sections/adr-examiner-notes-history-view/adr-examiner-notes-history-view.component';
import { AdrNewCertificateRequiredViewComponent } from './custom-sections/adr-new-certificate-required-view/adr-new-certificate-required-view.component';
import { AdrSectionEditComponent } from './custom-sections/adr-section/adr-section-edit/adr-section-edit.component';
import { AdrSectionSummaryComponent } from './custom-sections/adr-section/adr-section-summary/adr-section-summary.component';
import { AdrSectionViewComponent } from './custom-sections/adr-section/adr-section-view/adr-section-view.component';
import { AdrSectionComponent } from './custom-sections/adr-section/adr-section.component';
import { AdrTankDetailsInitialInspectionViewComponent } from './custom-sections/adr-tank-details-initial-inspection-view/adr-tank-details-initial-inspection-view.component';
import { AdrTankDetailsM145ViewComponent } from './custom-sections/adr-tank-details-m145-view/adr-tank-details-m145-view.component';
import { AdrTankDetailsSubsequentInspectionsEditComponent } from './custom-sections/adr-tank-details-subsequent-inspections-edit/adr-tank-details-subsequent-inspections-edit.component';
import { AdrTankDetailsSubsequentInspectionsViewComponent } from './custom-sections/adr-tank-details-subsequent-inspections-view/adr-tank-details-subsequent-inspections-view.component';
import { AdrTankStatementUnNumberEditComponent } from './custom-sections/adr-tank-statement-un-number-edit/adr-tank-statement-un-number-edit.component';
import { AdrTankStatementUnNumberViewComponent } from './custom-sections/adr-tank-statement-un-number-view/adr-tank-statement-un-number-view.component';
import { AdrComponent } from './custom-sections/adr/adr.component';
import { BodyComponent } from './custom-sections/body/body.component';
import { CustomDefectComponent } from './custom-sections/custom-defect/custom-defect.component';
import { CustomDefectsComponent } from './custom-sections/custom-defects/custom-defects.component';
import { CustomFormControlComponent } from './custom-sections/custom-form-control/custom-form-control.component';
import { DefectComponent } from './custom-sections/defect/defect.component';
import { DefectsComponent } from './custom-sections/defects/defects.component';
import { DimensionsComponent } from './custom-sections/dimensions/dimensions.component';
import { LettersComponent } from './custom-sections/letters/letters.component';
import { ModifiedWeightsComponent } from './custom-sections/modified-weights/modified-weights.component';
import { PlatesComponent } from './custom-sections/plates/plates.component';
import { PsvBrakesComponent } from './custom-sections/psv-brakes/psv-brakes.component';
import { RequiredStandardComponent } from './custom-sections/required-standard/required-standard.component';
import { RequiredStandardsComponent } from './custom-sections/required-standards/required-standards.component';
import { TrlBrakesComponent } from './custom-sections/trl-brakes/trl-brakes.component';
import { TyresComponent } from './custom-sections/tyres/tyres.component';
import { WeightsComponent } from './custom-sections/weights/weights.component';

@NgModule({
	declarations: [
		BaseControlComponent,
		TextInputComponent,
		ViewListItemComponent,
		DynamicFormGroupComponent,
		ViewCombinationComponent,
		CheckboxGroupComponent,
		RadioGroupComponent,
		DefectComponent,
		DefectsComponent,
		AutocompleteComponent,
		NumberInputComponent,
		TextAreaComponent,
		NumberOnlyDirective,
		ToUppercaseDirective,
		NoSpaceDirective,
		TrimWhitespaceDirective,
		DateComponent,
		DateControlsComponent,
		CharacterCountComponent,
		ControlErrorsComponent,
		SelectComponent,
		DynamicFormFieldComponent,
		FieldErrorMessageComponent,
		DefectSelectComponent,
		RequiredStandardSelectComponent,
		DateFocusNextDirective,
		TruncatePipe,
		WeightsComponent,
		LettersComponent,
		PlatesComponent,
		DimensionsComponent,
		TrlBrakesComponent,
		ReadOnlyComponent,
		CustomDefectsComponent,
		RequiredStandardComponent,
		RequiredStandardsComponent,
		CustomDefectComponent,
		SwitchableInputComponent,
		ReadOnlyComponent,
		SuffixDirective,
		AbandonDialogComponent,
		BodyComponent,
		TyresComponent,
		PsvBrakesComponent,
		PrefixDirective,
		SuggestiveInputComponent,
		CheckboxComponent,
		ApprovalTypeComponent,
		ApprovalTypeInputComponent,
		ApprovalTypeFocusNextDirective,
		ModifiedWeightsComponent,
		FieldWarningMessageComponent,
		AdrComponent,
		AdrTankDetailsSubsequentInspectionsEditComponent,
		AdrTankStatementUnNumberEditComponent,
		CustomFormControlComponent,
		AdrExaminerNotesHistoryEditComponent,
		AdrExaminerNotesHistoryViewComponent,
		AdrTankDetailsSubsequentInspectionsViewComponent,
		AdrTankDetailsInitialInspectionViewComponent,
		AdrTankStatementUnNumberViewComponent,
		AdrCertificateHistoryComponent,
		AdrTankDetailsM145ViewComponent,
		ContingencyAdrGenerateCertComponent,
		AdrNewCertificateRequiredViewComponent,
		AdrSectionComponent,
		AdrSectionEditComponent,
		AdrSectionViewComponent,
		AdrSectionSummaryComponent,
		AdrPermittedDangerousGoodsComponent,
	],
	imports: [CommonModule, FormsModule, ReactiveFormsModule, SharedModule, RouterModule],
	exports: [
		TextInputComponent,
		ViewListItemComponent,
		DynamicFormGroupComponent,
		ViewCombinationComponent,
		CheckboxGroupComponent,
		RadioGroupComponent,
		DefectComponent,
		DefectsComponent,
		AutocompleteComponent,
		NumberInputComponent,
		TextAreaComponent,
		DateComponent,
		DateControlsComponent,
		CharacterCountComponent,
		ControlErrorsComponent,
		SelectComponent,
		DynamicFormFieldComponent,
		FieldErrorMessageComponent,
		DefectSelectComponent,
		RequiredStandardSelectComponent,
		WeightsComponent,
		LettersComponent,
		PlatesComponent,
		TyresComponent,
		DimensionsComponent,
		TrlBrakesComponent,
		ReadOnlyComponent,
		RequiredStandardComponent,
		RequiredStandardsComponent,
		CustomDefectsComponent,
		CustomDefectComponent,
		SwitchableInputComponent,
		SuffixDirective,
		ReadOnlyComponent,
		AbandonDialogComponent,
		BodyComponent,
		PsvBrakesComponent,
		PrefixDirective,
		SuggestiveInputComponent,
		CheckboxComponent,
		ToUppercaseDirective,
		NoSpaceDirective,
		TrimWhitespaceDirective,
		ApprovalTypeComponent,
		ApprovalTypeInputComponent,
		ApprovalTypeFocusNextDirective,
		ModifiedWeightsComponent,
		AdrComponent,
		AdrCertificateHistoryComponent,
		FieldWarningMessageComponent,
		AdrSectionComponent,
		AdrSectionEditComponent,
		AdrSectionViewComponent,
		AdrSectionSummaryComponent,
	],
})
export class DynamicFormsModule {}
