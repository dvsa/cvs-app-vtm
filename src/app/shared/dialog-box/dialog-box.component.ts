import { Component, OnInit, ChangeDetectionStrategy, Inject } from '@angular/core';
import { MAT_DIALOG_DATA, MatDialogRef } from '@angular/material/dialog';

export interface DialogData {
  response: string;
  context: string;
  actionName: string;
}

@Component({
  selector: 'vtm-dialog-box',
  templateUrl: './dialog-box.component.html',
  styleUrls: ['./dialog-box.component.scss'],
  changeDetection: ChangeDetectionStrategy.OnPush
})
export class DialogBoxComponent implements OnInit {
  constructor(
    public dialogRef: MatDialogRef<DialogBoxComponent>,
    @Inject(MAT_DIALOG_DATA) public data: DialogData
  ) {}

  ngOnInit() {}

  close(): void {
    this.dialogRef.close(false);
  }

  save(modalData:  string): void {
    this.dialogRef.close({isSave: true, data: modalData});
  }
}
