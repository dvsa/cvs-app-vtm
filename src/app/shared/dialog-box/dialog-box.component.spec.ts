import { async, ComponentFixture, TestBed } from '@angular/core/testing';

import { DialogBoxComponent } from './dialog-box.component';
import { FormsModule } from '@angular/forms';
import {MAT_DIALOG_DATA, MatDialogModule, MatDialogRef} from '@angular/material/dialog';
import {MaterialModule} from '@app/material.module';

describe('DialogBoxComponent', () => {
  let component: DialogBoxComponent;
  let fixture: ComponentFixture<DialogBoxComponent>;
  const dialogMock = { close: () => {} };

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [DialogBoxComponent],
      imports: [MatDialogModule, MaterialModule, FormsModule],
      providers: [
        { provide: MatDialogRef, useValue: dialogMock },
        { provide: MAT_DIALOG_DATA, useValue: {} }
      ]
    }).compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(DialogBoxComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
    expect(fixture).toMatchSnapshot();
  });

  describe('close', () => {
    it('should close the modal when called', () => {
      spyOn(component.dialogRef, 'close');
      component.close();
      expect(component.dialogRef.close).toHaveBeenCalledWith(false);

    });
  });

  describe('save', () => {
    it('should close the modal & send modal data', () => {
      const modalData = { isSave: true, data: 'test data' };
      spyOn(component.dialogRef, 'close');
      component.save('test data');
      expect(component.dialogRef.close).toHaveBeenCalledWith(modalData);

    });
  });

});
