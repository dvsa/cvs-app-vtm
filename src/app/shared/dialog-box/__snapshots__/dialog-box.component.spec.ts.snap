// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DialogBoxComponent should create 1`] = `
<vtm-dialog-box
  data={[Function Object]}
  dialogRef={[Function Object]}
>
  <div
    class="mat-dialog-content"
    mat-dialog-content=""
  >
    <h2
      class="govuk-heading-l"
    >
      
    </h2>
    <textarea
      aria-describedby="reasonForCreation"
      class="govuk-textarea govuk-!-width-three-quarters ng-untouched ng-pristine ng-valid"
      id="test-reasonForCreation"
      rows="8"
    >
        
    </textarea>
  </div><button
    cdkfocusinitial=""
    class="govuk-button"
    data-module="govuk-button"
  >
     
  
  </button><a
    class="govuk-link govuk-!-margin-left-2"
    href="javascript:void(0)"
  >
    <span>
      Cancel
    </span>
  </a>
</vtm-dialog-box>
`;
